src/sio_client.cpp

```diff
@@ -12,8 +12,8 @@ using std::stringstream;
 
 namespace sio
 {
-    client::client():
-        m_impl(new client_impl())
+    client::client(client_options const& options):
+        m_impl(new client_impl(options))
     {
     }
```

Comment:
```suggestion
    client::client():
        m_impl(new client_impl(client_options{}))
    {
    }
    client::client(client_options const& options):
        m_impl(new client_impl(options))
    {
    }
```

Keep previous constructor for backwards compatibility